editor/editor_file_system.cpp

```diff
@@ -1974,15 +1974,17 @@ void EditorFileSystem::_update_script_documentation() {
 		for (int i = 0; i < ScriptServer::get_language_count(); i++) {
 			ScriptLanguage *lang = ScriptServer::get_language(i);
 			if (lang->supports_documentation() && efd->files[index]->type == lang->get_type()) {
-				bool should_reload_script = _should_reload_script(path);
 				Ref<Script> scr = ResourceLoader::load(path);
 				if (scr.is_null()) {
 					continue;
 				}
-				if (should_reload_script) {
-					// Reloading the script from disk. Otherwise, the ResourceLoader::load will
-					// return the last loaded version of the script (without the modifications).
-					scr->reload_from_file();
+				if (lang->get_name() != "C#") {
+					bool should_reload_script = _should_reload_script(path);
+					if (should_reload_script) {
+						// Reloading the script from disk. Otherwise, the ResourceLoader::load will
+						// return the last loaded version of the script (without the modifications).
+						scr->reload_from_file();
+					}
```

Comment:
These changes don't seem to be needed, so I would revert them.

Reply:
@magian1127 This still needs to be addressed. In my testing it didn't seem to be needed, but I could be missing something, so if you still think it's needed could you explain why? Thank you.

Reply:
Yeah, I'd hate to see code like this start getting added, especially if folks end up adding more scripting languages in the future.

We should eventually all get together and figure out what a good doc-generation architecture might be. It probably wants to life inside `Script`, `ScriptLanguage` and `ScriptServer`, which would abstract it away nicely and not require special cases!

Reply:
@magian1127 Thanks, I'm able to reproduce the exception. I understand why you made this change now.

Reloading a C# script checks if the script is already registered before trying to register it again. So if it still throws after that, it must mean the script is different, and that's a bug. There must always be only one `CSharpScript` instance for each C# type, so if another one is being created somewhere we need to find where.

Reply:
I haven't found any other solution yet.
been a little busy lately, so I'll look into it when I'm free.

Reply:
The potentially related PR #98269